server/src/main/java/io/crate/analyze/GeneratedColumnExpander.java

```diff
@@ -152,7 +149,8 @@ public Symbol visitFunction(Function function, Context context) {
 
 
         private Symbol addComparison(Function function, Reference reference, Symbol comparedAgainst, Context context) {
-            Collection<GeneratedReference> genColInfos = context.referencedRefsToGeneratedColumn.get(reference);
+            ArrayList<GeneratedReference> genColInfos = context.referencedRefsToGeneratedColumn
+                .getOrDefault(reference, new ArrayList<>());
```

Comment:
This always creates the `new ArrayList` even if it's not required.

Reply:
i probably could change `ArrayList<GeneratedReference>` to `List<..>` and then use `List.of()` in `getOrDefault`, but `computeIfAbsent` should be ok as well, not sure what is better